pyvista/core/filters/data_set.py

```diff
@@ -5180,6 +5194,267 @@ def extract_points(self, ind, adjacent_cells=True, include_cells=True, progress_
+        is called to extract cells for cell data.
+
+        By default, two arrays are included with the output: ``'vtkOriginalPointIds'``
+        and ``'vtkOriginalCellIds'``. These arrays can be used to link the filtered
+        points or cells directly to the input.
+
+        .. versionadded:: 0.44
+
+        Parameters
+        ----------
+        values : iterable[float | sequence[float, float]], optional
+            Iterable of point values to be extracted. Each item of the iterable may be
+            a single number or a sequence of two numbers defining a range of values
+            in the form``[lower, upper]``. Multiple single-number entries and/or
+            multiple ranges may be specified. If no values are specified, each unique
+            value in the data is extracted independently and returned as a
+            :class:`~pyvista.MultiBlock` dataset.
+
+            .. note::
+                Use ``+/-`` infinity when specifying range values for open
+                intervals, e.g.:
+
+                -   ``[0, float('inf')]`` to extract all values greater than
+                    or equal to zero
+                -   ``[float('-inf'), 0]`` to extract all values less than
+                    or equal to zero.
+
+        scalars : str, optional
+            Name of scalars to extract points with. Defaults to currently
+            active scalars.
+
+        preference : str, default: 'point'
+            When ``scalars`` is specified, this is the preferred array
+            type to search for in the dataset.  Must be either
+            ``'point'`` or ``'cell'``.
+
+        invert : bool, default: False
+            Invert the extraction values. If ``True`` extract the points (with
+            cells) which do *not* have the specified values.
+
+        adjacent_cells : bool, default: True
+            If ``True``, include cells (and their points) that contain
+            at least one of the extracted points. If ``False``, only
+            include cells which contain exclusively points from the
+            extracted points list. Has no effect if ``include_cells``
+            is ``False``. Has no effect when extracting values from
+            cell data.
+
+        include_cells : bool, default: True
+            Specifies if the cells shall be returned or not. If ``False``,
+            all cells in the output will be vertex cells, one for each point.
+            Has no effect when extracting values from cell data.
+
+        keep_original_ids : bool, default: True
+            If ``True``, two arrays ``'vtkOriginalPointIds'`` and ``'vtkOriginalCellIds'``
+            are included with the output. Otherwise, these arrays are cleared
+            from the output.
+
+        split : bool, default: False
+            If ``True``, each item in the ``values`` input is extracted
+            independently and a :class:`~pyvista.MultiBlock` is returned
+            with each extraction stored as a separate mesh.
```

Comment:
```suggestion
            with each extraction stored as a separate mesh. The number 
            of blocks returned equals the number of input values, i.e. 
            ``n_block == len(values)``
            
            .. note::
                Output blocks may contain empty meshes if any specified ``values``
                do not exist in the mesh. This can impact plotting since empty meshes
                cannot be plotted by default. Use :meth:`pyvista.MultiBlock.clean` on 
                the output to remove empty meshes, or set ``pv.global_theme.allow_empty_mesh = True``
                to enable plotting empty meshes.

```